community/kernel/src/test/java/org/neo4j/unsafe/impl/batchimport/cache/ByteArrayTest.java

```diff
@@ -20,34 +20,64 @@
 package org.neo4j.unsafe.impl.batchimport.cache;
 
 import org.junit.After;
+import org.junit.AfterClass;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.Parameterized;
 import org.junit.runners.Parameterized.Parameter;
 import org.junit.runners.Parameterized.Parameters;
 
+import java.io.File;
+import java.io.IOException;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.concurrent.ThreadLocalRandom;
 import java.util.function.Supplier;
 
+import org.neo4j.io.pagecache.PageCache;
+
+import static org.junit.Assert.assertArrayEquals;
 import static org.junit.Assert.assertEquals;
+import static org.neo4j.unsafe.impl.batchimport.cache.NumberArrayFactory.AUTO;
+import static org.neo4j.unsafe.impl.batchimport.cache.NumberArrayFactory.HEAP;
+import static org.neo4j.unsafe.impl.batchimport.cache.NumberArrayFactory.OFF_HEAP;
+import static org.neo4j.unsafe.impl.batchimport.cache.NumberArrayFactory.autoWithPageCacheFallback;
 
 @RunWith( Parameterized.class )
-public class ByteArrayTest
+public class ByteArrayTest extends NumberArrayPageCacheTestSupport
 {
     private static final byte[] DEFAULT = new byte[25];
+    private static final int LENGTH = 1_000;
+    private static Fixture fixture;
 
     @Parameters
-    public static Collection<Supplier<ByteArray>> data()
+    public static Collection<Supplier<ByteArray>> data() throws IOException
     {
+        fixture = prepareDirectoryAndPageCache( ByteArrayTest.class );
```

Comment:
This is needed due to some unfortunate interaction between class rules, parameterized tests and testdirectories, IIRC.